src/main.c

```diff
@@ -21,10 +21,35 @@ FailureMode find_failure_mode(char *file_path, char *file_name_no_ext);
 bool is_file_diff(Exercise *exercise);
 
 static void sig_handler(int _);
+void print_usage();
 
-int main() {
+int main(int argc, char *argv[]) {
     signal(SIGINT, sig_handler);
 
+    int start_at = 1;
+    int c;
+
+    opterr = 0;
+
+    while ((c = getopt(argc, argv, "s::")) != -1) {
+        switch (c) {
+            case 's':
+                start_at = atoi(optarg);
```

Comment:
Could you add a check here to make sure optarg isn't null? if it is null, then we should display some message that the -s requires a number to start at